cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/AdaptTemplatesCommand.java

```diff
@@ -0,0 +1,239 @@
+                if (Files.notExists(cobigenTemplatesFolderPath)) {
+                    Files.createDirectory(cobigenTemplatesFolderPath);
+                }
+                while (entries.hasMoreElements()) {
+                    ZipEntry entry = entries.nextElement();
+                    if (entry.getName().equals("pom.xml")) {
+                        Path saveForFileCreationPath =
+                            fileSystem.getPath(templatesFolderPath + File.separator + entry.getName());
+                        createFile(file, entry, saveForFileCreationPath);
+                    }
+                }
+            } catch (IOException e) {
+                throw new IOException(
+                    "An exception occurred while unpacking pom.xml from Jar file to templates folder");
+            }
+        }
+
+        // unpack sources
+        try (ZipFile file = new ZipFile(sourcesJarPath)) {
+            Enumeration<? extends ZipEntry> entries = file.entries();
+            if (Files.notExists(cobigenTemplatesFolderPath)) {
+                Files.createDirectory(cobigenTemplatesFolderPath);
+            }
+            while (entries.hasMoreElements()) {
+                ZipEntry entry = entries.nextElement();
+                Path saveForFileCreationPath =
+                    fileSystem.getPath(templatesFolderPath + File.separator + entry.getName());
+                if (entry.getName().contains("context.xml")) {
+                    saveForFileCreationPath =
+                        fileSystem.getPath(templatesFolderPath + File.separator + entry.getName());
+                } else if (entry.getName().contains("com/")) {
+                    saveForFileCreationPath = fileSystem.getPath(templatesFolderPath + File.separator + "src"
+                        + File.separator + "main" + File.separator + "java" + File.separator + entry.getName());
+                }
+                createFile(file, entry, saveForFileCreationPath);
+            }
+        } catch (IOException e) {
+            throw new IOException("An exception occurred while unpacking sources from Jar file to templates folder");
+        }
+
+        // unpack classes to target directory
+        try (ZipFile file = new ZipFile(classesJarPath)) {
+            Enumeration<? extends ZipEntry> entries = file.entries();
+            Path sourcesClassPath = fileSystem.getPath(templatesFolderPath + File.separator + "target");
+            if (Files.notExists(sourcesClassPath)) {
+                Files.createDirectory(sourcesClassPath);
+            }
+            while (entries.hasMoreElements()) {
+                ZipEntry entry = entries.nextElement();
+                if (entry.getName().contains("com/")) {
+                    Path saveForFileCreationPath = fileSystem.getPath(sourcesClassPath + File.separator + "src"
+                        + File.separator + "main" + File.separator + "java" + File.separator + entry.getName());
```

Comment:
please use the Path.resolve syntax to not fiddle with File.separator appends

Reply:
adjusted.